### PR TITLE
Fix flow cancellation for containerized dispatch

### PR DESCRIPTION
**Pull request by sshardool**

### CODE REVIEW

File: azkaban-common/src/main/java/azkaban/executor/container/ContainerizedDispatchManager.java [hunk 1]
```diff
@@ -320,6 +323,25 @@ public void run() {
...
+      final Map<Integer, Pair<ExecutionReference, ExecutableFlow>> unfinishedFlows = this.executorLoader
+          .fetchUnfinishedFlows();
+      if (unfinishedFlows.containsKey(exFlow.getExecutionId())) {
+        final Pair<ExecutionReference, ExecutableFlow> pair = unfinishedFlows
+            .get(exFlow.getExecutionId());
+        // Note that ExecutionReference may have the 'executor' as null. ApiGateway call is expected
+        // to handle this scenario.
+        this.apiGateway.callWithReferenceByUser(pair.getFirst(), ConnectorParams.CANCEL_ACTION, userId);
+      } else {
+        throw new ExecutorManagerException("Execution "
```
Comment (jakhani) [inline azkaban-common/src/main/java/azkaban/executor/container/ContainerizedDispatchManager.java:339]: Can you please add error log over here? I also miss this alot :-)

File: azkaban-common/src/main/java/azkaban/executor/container/ContainerizedDispatchManager.java [hunk 2]
```diff
@@ -320,6 +323,25 @@ public void run() {
     }
   }
 
+  @Override
+  public void cancelFlow(ExecutableFlow exFlow, String userId)
```
Comment (jakhani) [inline azkaban-common/src/main/java/azkaban/executor/container/ContainerizedDispatchManager.java:327]: What's the difference between method in AbstractExecutorManagerAdapter and this?

Comment (sshardool) [reply azkaban-common/src/main/java/azkaban/executor/container/ContainerizedDispatchManager.java:327]: The implementation in `AbstractExecutorManagerAdapter` explicitly relies on the existence of a non-null `executor` in the ExecutionReference which result in the failure of cancelFlow. For ContainerizedDispatch a valid executor is not required and will typically not be present. (If it does exist for any reason, ApiGateway calls have already been modified to handle it correctly)

**Review by jakhani — APPROVED** [review]

File: azkaban-common/src/main/java/azkaban/executor/container/ContainerizedDispatchManager.java (hunk 1 above)
Comment (sshardool) [reply azkaban-common/src/main/java/azkaban/executor/container/ContainerizedDispatchManager.java]: Fixed.

**Pull request merged**